### DISCUSSION TITLE
extracting values from inference results

### DISCUSSION THREAD

**Discussion by Finfalter** [sonos/tract#593] [Q&A] [answered]
Could you sketch the way how to further process the inference (extract the result) from a model? To explain what I mean: I took the model of tensorflow's [hello_world](https://github.com/tensorflow/tflite-micro/tree/main/tensorflow/lite/micro/examples/hello_world) example and got some results, for example

`[[0.8327713]], shape=[1, 1], strides=[1, 1], layout=CFcf (0xf), dynamic ndim=2`

The underlying model is supposed to have learned `sine()`. The input in this case was  0.94529176, so the output is ok with inferred 0.8327713. The underlying variable of the displayed value is of type

`ndarray::ArrayBase<ndarray::ViewRepr<&f32>, ndarray::dimension::dim::Dim<ndarray::dimension::dynindeximpl::IxDynImpl>>`

I can display the entire output as shown above but in order to further process it, I have to extract it as a `f32`. 
How would I do that?

**Comment by kali** [comment, marked as answer]
Hey!

The array_view methods, like you tried, are a bridge to ndarray ArrayBase. You can access values by indexing (using []) like shown here: https://docs.rs/ndarray/0.15.4/ndarray/struct.ArrayBase.html#indexing-and-dimension.

**Reply by Finfalter** [reply]
It works, thanks a lot!
